File: src/Algorithm/golang/35. Search Insert Position.go
func searchInsert(nums []int, target int) int {
    var left,right int = 0,len(nums)-1;
    for left<right {
        var mid int = (left+right)/2;
        if(nums[mid] == target){
            return mid;
        }else if (nums[mid] < target){
            left = mid+1;
        }else{
            right = mid-1;
        }
    }
    if(nums[left] < target){
        return left+1;
    }
    return left;
}

File: src/Algorithm/java/35. Search Insert Position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int start = 0;
        int end = nums.length-1;
        while(start<end){
            int mid = (int)((long)start+(long)end)/2;
            if(nums[mid] == target){
                return mid;
            }else if(nums[mid]<target){
                start = mid+1;
            }else if(nums[mid]>target){
                end = mid-1;
            }
            
        }
        return nums[start]<target?start+1:start;
    }
}